spec/std/file_spec.cr

```diff
@@ -754,7 +785,7 @@ describe "File" do
   it "opens with perm (int)" do
     with_tempfile("write_with_perm-int.txt") do |path|
       perm = 0o600
-      File.open(path, "w", perm) do |file|
+      File.open(path, :write, :create, permissions: perm) do |file|
```

Comment:
I think we are forcing named arguments in order to overcome a limitation of autocasting of enums flags.

If the mode enum covers the more common combination like `:overwrite` then I would prefer to remove the named argument in favor of the type-safe arguments only. 

From the `File::Mode.flags(Create, Append)` and this maybe we need another enum value that can be used plainly as `:write_create` or `:w`

Reply:
I think the `*args` interface is the neatest. Passing custom flags like `:write, :create` seems more common than passing custom create permissions.

That being said, there might be a better interface I can't think of right now.

Reply:
I think I prefer to

* cover the `write|create`  and `creat|append` with additional enum values
* say with a single positional mode argument
* force longer construct as in File.write spec if those are not covered in the enum

This will leave similar api on File.write and File.open since due to the default arguments in the former there can't be a varyadic mode.

Reply:
I still think this is optimizing for the wrong thing. I only added `Overwrite` since it was a *lot* shorter and it was easy to name. It's obviously not feasible to add one for every combination, and I'd like to keep the API nice for those.

Empirical data would be the use rate of `permissions`. Since we havent supported these more specific combinations before, there's no empirical data on their use rate unfortunately.